solution_runner/commands/setup_command.py

```diff
@@ -0,0 +1,129 @@
+from datetime import datetime
+from pathlib import Path
+
+import click
+import requests
+
+from . import consts, commands_utils
+from .consts import Directories, FileExtensions
+from .defaults_and_choices import get_default_year
+
+
+_default_year = get_default_year()
+
+
+@click.command(name='setup')
+@click.option('-y', '--year', type=click.IntRange(consts.FIRST_AOC_YEAR, _default_year), default=_default_year,
+              show_default=f'last year: {_default_year}', help='year of puzzle setting up solution for')
+@click.option('-d', '--day', type=consts.ADVENT_DAYS_RANGE, required=True, help='day of puzzle setting up solution for')
+@click.option('--use_cache/--ignore_cache', 'should_use_cache', default=True, show_default='true',
+              help='whether to use cached input file')
+def command(year: int, day: int, should_use_cache: bool):
+    """Set up a solution: fetch input and create solution files."""
+    _abort_if_puzzle_locked(year, day)
+
+    year = str(year)
+    day = str(day).zfill(2)  # Add a leading zero for single digit numbers.
+
+    try:
+        root_directory = commands_utils.get_setting(consts.ROOT_DIRECTORY)
+    except FileNotFoundError:
+        click.Context(command).abort()
+    inputs_directory = root_directory / Directories.INPUTS
+    _ask_user_to_mkdir(inputs_directory, 'input files')
+
+    year_inputs_directory = inputs_directory / year
+    _ask_user_to_mkdir(year_inputs_directory, f'{year} input files')
+    input_file = (year_inputs_directory / day).with_suffix(FileExtensions.TEXT)
+
+    if should_use_cache and input_file.exists() and input_file.read_text():  # Abort if the file exists but it's empty.
+        _abort_input_file_already_exists(year, day)
+    input_file.touch()
```

Comment:
Why?

Reply:
True, this is redundant. Removed in 8c3375ec.